app/jobs/response_csv_export_operation_job.rb

```diff
@@ -0,0 +1,30 @@
+class ResponseCsvExportOperationJob < OperationJob
+  def perform(operation, mission, search = nil)
+    ability = Ability.new(user: operation.creator, mission: mission)
+    responses = Response.accessible_by(ability, :export)
+
+    # do search, excluding excerpts
+    if search.present?
+      responses = Response.do_search(
+        responses,
+        search,
+        {mission: mission},
+        include_excerpts: false
+      )
+    end
+
+    # Get the response, for export, but not paginated.
+    # We deliberately don't eager load as that is handled in the Results::Csv::Generator class.
+    responses = responses.order(:created_at)
+
+    csv = Results::Csv::Generator.new(responses)
```

Comment:
Could this just be `attachment = Results::Csv::Generator.new(responses).export`, and you return the file itself from the export method?